cmake/cyclonedds/CMakeLists.txt

```diff
@@ -44,9 +40,15 @@ function(fetch_and_install name)
     set(BUILD_DIR ${CMAKE_BINARY_DIR}/dependencies/${name}/build)
     set(INSTALL_DIR ${CMAKE_BINARY_DIR}/dependencies/install)
 
+    set(CYCLONE_DOWNLOAD_CMD COMMAND ${CMAKE_COMMAND} ${ENABLE_STATIC_DEBUG} -G "${CMAKE_GENERATOR}" "${DOWNLOAD_CONFIG_DIR}")
+
+    if(DEFINED CMAKE_TOOLCHAIN_FILE)
+        list(APPEND CYCLONE_DOWNLOAD_CMD "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
+    endif()
+
     # Fetch source
     configure_file(${name}.cmake.in ${DOWNLOAD_CONFIG_DIR}/CMakeLists.txt)
-    execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "${TOOLCHAIN_FILE}" "${DOWNLOAD_CONFIG_DIR}"
+    execute_process(${CYCLONE_DOWNLOAD_CMD}
```

Comment:
I'm not sure if this just works by accident?
Why not just using `EXTRA_CMAKE_ARGS`
```cmake
list(APPEND EXTRA_CMAKE_ARGS "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
```
And use this variable in all `${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"` commands

Reply:
@elBoberido 
The `EXTRA_CMAKE_ARGS` are initiallized with the `CMAKE_CXX_FLAGS` that are not used when downloading cyclonedds first. This would  drop a warning like this:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_FLAGS
```

Reply:
Okay. How about a bit more refactoring then. Let's crate a `EXTRA_CMAKE_CXX_FLAGS` and use `EXTRA_CMAKE_ARGS` for this?

If you are busy, I could take care of if.